pkg/server/headers/oauthbasic.go

```diff
@@ -3,25 +3,35 @@ package headers
 import "net/http"
 
 const (
-	authzHeader     = "Authorization"
-	copyAuthzHeader = "oauth.openshift.io:" + authzHeader // will never conflict because : is not a valid header key
+	headerCopyPrefix = "oauth.openshift.io:" // will never conflict because : is not a valid header key
 )
 
-func WithPreserveAuthorizationHeader(handler http.Handler) http.Handler {
+var oauthHeaders = []string{
+	"Authorization",
+	"X-Remote-User",
```

Comment:
Should we keep all the authentication headers the default handler chain removes?
The requestheader provider won't work properly in case any of the headers configured in https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L415-L424 match any headers that are being removed in the kube default handler chain.